solutions/usaco-922.mdx

```diff
@@ -9,7 +9,9 @@ author: Chongtian Ma
 
 ## Alternate Solution - Binary Search
 
-We can binary search for the longest possible prefix. Prefix of length $i-1$ is just prefix of length $i$ without a plate; if is it possible to wash all plates of prefix $i$ then it must also be possible to wash a prefix of $i-1$. Now we only need to simulate the stacking and washing. As the official analysis outlines, if plate $A$ is on top of plate $B$ on the same stack, then $A$ must be less than $B$, or else we cannot access the plate beneath it. Thus, we can greedily place each plate on stack $i$ if the top plate on stack $i$ is larger than our current plate and is the smallest out of all the other soapy stacks. This is because it is always more optimal to satisfy a stricter constraint, rather overriding a broader one if we place the plate on another stack which the plate on top is not the smallest possible.
+We can binary search for the longest possible prefix. Prefix of length $i-1$ is just prefix of length $i$ without a plate; if is it possible to wash all plates of prefix $i$ then it must also be possible to wash a prefix of $i-1$. Now we only need to simulate the stacking and washing. As the official analysis outlines, if plate $A$ is on top of plate $B$ on the same stack, then $A$ must be less than $B$, or else we cannot access the plate beneath it. 
```

Comment:
> Prefix of length $i-1$ is just prefix of length $i$ without a plate;

grammar

Reply:
wait whats wron gwith it lol

Reply:
it's a fragment?

Reply:
wha can u just fix it for me lol

Reply:
no.

Reply:
ill just assume nothing is wrong then unless someone says otherwise 😄

Reply:
how is a fragment not wrong

Reply:
am i blind wheres the fragment

Reply:
https://en.wikipedia.org/wiki/Semicolon

> a semicolon is most commonly used to link (in a single sentence) two independent clauses that are closely related in thought

> Prefix of length $i-1$ is just prefix of length $i$ without a plate;

you tell me how this is an independent clause

Reply:
"Prefix" is the subject, "is" is the verb? Do you want me to add "A" in front of it or something

Reply:
oh yeah, mb mb
what i meant is "Prefix" being the subject is very very awkward

Reply:
not addressed

Reply:
fine ill add your "A"